cg/server/endpoints/applications.py

```diff
@@ -23,6 +20,14 @@ def get_applications():
     return jsonify(applications=parsed_applications)
 
 
+@APPLICATIONS_BLUEPRINT.route("/applications/<order_type>")
+def get_application_order_types(order_type: str):
+    """Return application order types.."""
+    applications: list[Application] = db.get_applications_by_ordertype(order_type)
+    parsed_applications: list[dict] = [application.to_dict() for application in applications]
+    return jsonify(applications=parsed_applications)
```

Comment:
add error handling for when no application tags can be found for the ordertype

Reply:
Do we consider this relevant even if we change the order_type parameter to an enum? To me, being able to parse the order type to the enum fulfills a similar purpose

Reply:
Changing order type to an enum would make the check wheter the order type is supported at all. That it is supported by the back end does not immediatly mean that there are also app tags for it in the database. One could argue that there always should be. However, it might get missed with the next implementation which would lead to non sense errors in the front end.

Reply:
If we have already checked the input and deemed it okay (as one can interpret being able to parse it into an enum) I think it could be considered normal and simply the case that there are no applications for this order type configured yet.

Reply:
I am fine with either - maybe it is just a matter of interpretation whether we consider "no applications found" an error or not. I do think that the type hint should reuse the OrderType enum though

Reply:
Yes what would happen when you get an ordertype for which the applications are not yet configured? The endpoint would return nothing for a field that is required and thus cannot be filled. Right?

Reply:
Would the response not just look like

```Json
{
    "applications": []
}
```
which is sort of fine by me - it would result in a drop down with no entries in the front end. Kind of correct but maybe confusing?

Reply:
Yes, as a user getting a drop down menu that is empty without indication that something is wrong is not a great experience in my opinion.

Reply:
Will resolve this in favour of a new discussion